pkgs/development/tools/knightos/genkfs/default.nix

```
@@ -0,0 +1,28 @@
+{ stdenv, fetchFromGitHub, gcc, cmake, asciidoc, libxslt }:
+
+
+stdenv.mkDerivation rec {
+  pname = "genkfs";
+
+  version = "1.3.2";
+
+  src = fetchFromGitHub {
+    owner = "KnightOS";
+    repo = "genkfs";
+    rev = version;
+    sha256 = "0f50idd2bb73b05qjmwlirjnhr1bp43zhrgy6z949ab9a7hgaydp";
+  };
+
+  nativeBuildInputs = [ cmake gcc ];
+
+  buildInputs = [ asciidoc libxslt.bin ];
```

```suggestion
  nativeBuildInputs = [ asciidoc cmake gcc ];
```

And use `asciidoc = asciidoc-full;` in the attrset passed to `callPackage`.

Is the `gcc` dependency necessary? Doesn't kfs compile with clang?

---

It seems that `asciidoc-full` is a much heavier dependency than `asciidoc`, can't we use `asciidoc + libxslt` instead?

---

That's a fair point, but it's only a build-time dependency, so it doesn't increase the transitive closure of the output.